### DISCUSSION TITLE
[Solved] Xdebug not working

### DISCUSSION THREAD

**Discussion by kossa** [beyondcode/herd-community#390] [Q&A] [answered]
Hello

I'm using the latest version of herd(1.4.0 currently)

I followed the instructions [here](https://herd.laravel.com/docs/1/advanced-usage/xdebug) but when I tried the restart herd I got  this error:
```
Xdebug: [Step Debug] Could not connect to debugging client. Tried: localhost:9003 (through xdebug.client_host/xdebug.client_port).
```

PHP.ini
```ini
zend_extension=/Applications/Herd.app/Contents/Resources/xdebug/xdebug-82-arm64.so
xdebug.mode=debug,develop
xdebug.start_with_request=yes
xdebug.start_upon_error=yes
```

![image](https://github.com/beyondcode/herd-community/assets/1175584/0210b315-9a68-4ac7-9f83-b36897db6006)

Even if in PHP info seems like xdebug is active

![image](https://github.com/beyondcode/herd-community/assets/1175584/517a004d-d3f7-4ec7-b1d1-6cc6b3ab91c6)

Running tests with pest coverage

<img width="1353" alt="image" src="https://github.com/beyondcode/herd-community/assets/1175584/cd4bb39f-c599-458d-a456-147f9520730a">

**Comment by mpociot** [comment]
If you want to add code coverage support, you should add "coverage" to the Xdebug mode.

**Reply by kossa** [reply]
Right the coverage is working fine, but getting many errors:
```
Xdebug: [Step Debug] Could not connect to debugging client. Tried: localhost:9003 (through xdebug.client_host/xdebug.client_port).
```

**Reply by mpociot** [reply]
Right. It seems like Xdebug is trying to connect to an IDE, which does not work. If you're using PHPStorm, it should allow incoming connections on the debug port.

**Reply by kossa** [reply]
I see, no I'm using VsCode :)

**Reply by kossa** [reply, marked as answer]
Ohh I fixed it by adding: `xdebug.log_level=0` source https://stackoverflow.com/questions/65059303/xdebug-could-not-connect-to-debugging-client-tried-localhost9000

**Reply by cAstraea** [reply]
@kossa I'm not sure this is the solution though ? It's just silencing the error so it doesn't spam everywhere with [Step Debug] Could not connect to debugging client. Tried: localhost:9003 (through xdebug.client_host/xdebug.client_port).
